src/Nethermind/Nethermind.Evm/Instruction.cs

```diff
@@ -174,18 +177,202 @@ public enum Instruction : byte
     public static class InstructionExtensions
     {
-        public static string? GetName(this Instruction instruction, bool isPostMerge = false, IReleaseSpec? spec = null) =>
-            instruction switch
+        public static int GetImmediateCount(this Instruction instruction, bool IsEofContext, byte jumpvCount = 0)
+            => instruction switch
+            {
+                Instruction.CALLF or Instruction.JUMPF => IsEofContext ? EvmObjectFormat.TWO_BYTE_LENGTH : 0,
+                Instruction.DUPN or Instruction.SWAPN => IsEofContext ? EvmObjectFormat.ONE_BYTE_LENGTH : 0,
+                Instruction.RJUMP or Instruction.RJUMPI => IsEofContext ? EvmObjectFormat.TWO_BYTE_LENGTH : 0,
+                Instruction.RJUMPV => IsEofContext ? jumpvCount * EvmObjectFormat.TWO_BYTE_LENGTH + EvmObjectFormat.ONE_BYTE_LENGTH : 0,
+                >= Instruction.PUSH0 and <= Instruction.PUSH32 => instruction - Instruction.PUSH0,
+                _ => 0
+            };
+        public static bool IsTerminating(this Instruction instruction) => instruction switch
+        {
+            Instruction.RETF or Instruction.INVALID or Instruction.STOP or Instruction.RETURN or Instruction.REVERT => true,
+            Instruction.JUMPF or Instruction.RETURNCONTRACT => true,
+            // Instruction.SELFDESTRUCT => true
+            _ => false
+        };
+
+        public static bool IsValid(this Instruction instruction, bool IsEofContext)
+        {
+            if (!Enum.IsDefined(instruction))
+            {
+                return false;
+            }
+
+            return instruction switch
+            {
+                Instruction.CALLF or Instruction.RETF or Instruction.JUMPF => IsEofContext,
+                Instruction.DUPN or Instruction.SWAPN or Instruction.EXCHANGE => IsEofContext,
+                Instruction.RJUMP or Instruction.RJUMPI or Instruction.RJUMPV => IsEofContext,
+                Instruction.CALL => !IsEofContext,
+                Instruction.CALLCODE => !IsEofContext,
+                Instruction.DELEGATECALL => !IsEofContext,
+                Instruction.SELFDESTRUCT => !IsEofContext,
+                Instruction.JUMP => !IsEofContext,
+                Instruction.JUMPI => !IsEofContext,
+                Instruction.PC => !IsEofContext,
+                Instruction.CREATE2 or Instruction.CREATE => !IsEofContext,
+                Instruction.CODECOPY => !IsEofContext,
+                Instruction.CODESIZE => !IsEofContext,
+                Instruction.EXTCODEHASH => !IsEofContext,
+                Instruction.EXTCODECOPY => !IsEofContext,
+                Instruction.EXTCODESIZE => !IsEofContext,
+                Instruction.GAS => !IsEofContext,
+                _ => true
+            };
+        }
+
+        //Note() : Extensively test this, refactor it, 
+        public static (ushort? InputCount, ushort? OutputCount, ushort? immediates) StackRequirements(this Instruction instruction) => instruction switch
+        {
```

Comment:
I dont like this, and still trying to find a better way to do it and encode the idea that some opcodes have Variable Length immediate arguments like JUMPV